sds_data_manager/lambda_code/SDSCode/instruments/l1b_Codice.py

```diff
@@ -0,0 +1,37 @@
+"""Lambda runtime code that triggers off of arrival of data into S3 bucket.
+"""
+import os
+from datetime import datetime
+
+import boto3
+
+#TODO: ability to access database, EFS, calibration data, etc.
+
+def lambda_handler(event: dict, context):
+    """Handler function"""
+    print(event)
+    print(context)
+
+    now = datetime.now()
+    print("Now time is")
+    print(now)
+
+    # Get the environment variables
+    bucket = os.environ['PROCESSING_PATH']
+    prefix = os.environ['INSTRUMENT_SOURCES']
+
+    # Retrieves objects in the S3 bucket under the given prefix
+    try:
+        s3 = boto3.client('s3')
+        object_list = s3.list_objects_v2(Bucket=bucket, Prefix=prefix)["Contents"]
```

Comment:
Is the end goal for this to confirm that the files needed for this level of processing are returned in the object list? Or will that be handled by event bridge?

Reply:
We will do the following:
1. Trigger the step function using EventBridge when new objects are created in s3 bucket or EFS
2. Each instrument will check to see if enough data is available for processing using the lambda (this is where the trigger gets refined)
3. Based on the results of the lambda the Batch Job will run

This will be worked out with the EventBridge code I'll work on next.